doc/sphinx/Pacemaker_Administration/pcs-crmsh.rst

```diff
@@ -0,0 +1,441 @@
+
+   .. code-block:: none
+
+      crmsh # crm configure
+      crmsh # edit
+      crmsh # verify
+      crmsh # commit
+
+.. topic:: Delete a Resource's Instance Parameters
+
+   .. code-block:: none
+
+      crmsh     # crm resource param ClusterIP delete nic
+      pcs       # pcs resource update ClusterIP nic=  
+      pacemaker # crm_resource -r ClusterIP --delete-parameter nic
+
+.. topic:: List Current Resource Defaults
+
+   .. code-block:: none
+
+      crmsh     # crm configure show type:rsc_defaults
+      pcs       # pcs resource defaults
+      pacemaker # cibadmin -Q --scope rsc_defaults
+
+.. topic:: Set Resource Defaults
+
+   .. code-block:: none
+
+      crmsh # crm configure rsc_defaults resource-stickiness=100
+      pcs   # pcs resource defaults resource-stickiness=100
+    
+.. topic:: List Current Operation Defaults
+
+   .. code-block:: none
+
+      crmsh     # crm configure show type:op_defaults
+      pcs       # pcs resource op defaults
+      pacemaker # cibadmin -Q --scope op_defaults
+
+.. topic:: Set Operation Defaults
+
+   .. code-block:: none
+
+      crmsh # crm configure op_defaults timeout=240s
+      pcs   # pcs resource op defaults timeout=240s
+
+.. topic:: Enable Resource Agent Tracing for a Resource
+
+   .. code-block:: none
+
+      crmsh # crm resource trace Website
+
+.. topic:: Clear Fail Counts for a Resource
+
+   .. code-block:: none
+
+      crmsh     # crm resource cleanup Website
+      pcs       # pcs resource cleanup Website
+      pacemaker # crm_resource --cleanup -r Website
+
+.. topic:: Create a Clone Resource
+
+   .. code-block:: none
+
+      crmsh # crm configure clone WebIP ClusterIP meta globally-unique=true clone-max=2 clone-node-max=2
+      pcs   # pcs resource clone ClusterIP globally-unique=true clone-max=2 clone-node-max=2
+
+.. topic:: Create a Promotable Clone Resource
+
+   .. code-block:: none
+
+      crmsh    # crm configure ms WebDataClone WebData \
+                 meta master-max=1 master-node-max=1 \
+                 clone-max=2 clone-node-max=1 notify=true
+      pcs-0.9  # pcs resource master WebDataClone WebData \
+                 master-max=1 master-node-max=1 \
+                 clone-max=2 clone-node-max=1 notify=true
+      pcs-0.10 # pcs resource promotable WebData WebDataClone \
+                 promoted-max=1 promoted-node-max=1 \
+                 clone-max=2 clone-node-max=1 notify=true
+
+pcs will generate the clone name automatically if it is omitted from the
+command line.
+
+
+Manage Constraints
+##################
+
+.. topic:: Create a Colocation Constraint
+
+   .. code-block:: none
+
+      crmsh # crm configure colocation website-with-ip INFINITY: WebSite ClusterIP
+      pcs   # pcs constraint colocation add ClusterIP with WebSite INFINITY
+
+.. topic:: Create a Colocation Constraint Based on Role
+
+   .. code-block:: none
+
+      crmsh # crm configure colocation another-ip-with-website inf: AnotherIP WebSite:Promoted
```

Comment:
It seems crmsh currently not support `Promoted` role, should still `Master` here

Reply:
OK, thanks. I guess crmsh restricts the user to certain values rather than pass them directly to Pacemaker?